modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web-test/src/testIntegration/java/com/liferay/portal/search/tuning/synonyms/web/internal/upgrade/v1_0_1/test/SynonymsConfigurationUpgradeProcessTest.java

```diff
@@ -0,0 +1,156 @@
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testUpgradeSynonymsConfiguration() throws Exception {
+		try {
+			_addConfiguration(_PID);
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _UPGRADE_VERSION);
+
+			upgradeProcess.upgrade();
+
+			_assertConfiguration(_PID);
+		}
+		finally {
+			_removeConfiguration(_PID);
+		}
+	}
+
+	private void _addConfiguration(String servicePid) throws Exception {
+		UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
+			new UnsyncByteArrayOutputStream();
+
+		ConfigurationHandler.write(
+			unsyncByteArrayOutputStream,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"filterNames", _FILTER_NAMES
+			).build());
+
+		try (Connection connection = DataAccess.getConnection();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				"insert into Configuration_ (configurationId, dictionary) " +
+					"values(?, ?)")) {
+
+			preparedStatement.setString(1, servicePid);
+			preparedStatement.setString(
+				2, unsyncByteArrayOutputStream.toString());
+
+			preparedStatement.execute();
+		}
+	}
+
+	private void _assertConfiguration(String servicePid) throws Exception {
+		try (Connection connection = DataAccess.getConnection();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"select dictionary from Configuration_ where ",
+					"configurationId = '", servicePid, "'"));
+			ResultSet resultSet = preparedStatement.executeQuery()) {
+
+			while (resultSet.next()) {
+				String dictionaryString = resultSet.getString("dictionary");
+
+				Dictionary<String, Object> dictionary =
+					ConfigurationHandler.read(
+						new UnsyncByteArrayInputStream(
+							dictionaryString.getBytes(StringPool.UTF8)));
+
+				String[] filterNameValues = (String[])dictionary.get(
+					"filterNames");
+
+				Assert.assertArrayEquals(
+					_EXPECTED_FILTER_NAMES, filterNameValues);
+			}
+		}
+	}
+
+	private void _removeConfiguration(String serviceFactoryPid)
+		throws Exception {
+
+		DB db = DBManagerUtil.getDB();
+
+		db.runSQL(
+			"delete from Configuration_ where configurationId like '" +
+				serviceFactoryPid + "%'");
+	}
+
+	private static final String[] _EXPECTED_FILTER_NAMES = {
+		"custom_filter_synonym_ru", "liferay_filter_synonym_ar",
+		"liferay_filter_synonym_ca", "liferay_filter_synonym_de",
+		"liferay_filter_synonym_en", "liferay_filter_synonym_es",
+		"liferay_filter_synonym_fi", "liferay_filter_synonym_fr",
+		"liferay_filter_synonym_hu", "liferay_filter_synonym_it",
+		"liferay_filter_synonym_ja", "liferay_filter_synonym_nl",
+		"liferay_filter_synonym_pt_BR", "liferay_filter_synonym_pt_PT",
+		"liferay_filter_synonym_sv", "liferay_filter_synonym_zh"
+	};
+
+	private static final String[] _FILTER_NAMES = {
+		"liferay_filter_synonym_en", "liferay_filter_synonym_es",
+		"custom_filter_synonym_ru"
+	};
+
+	private static final String _PID =
```

Comment:
I think remove this from being passed in the 3 methods and just call it globally.

Also rename to _CONFIGURATION_ID.